ui-v2/app/routes/dc/kv/create.js

```diff
@@ -3,14 +3,18 @@ import { inject as service } from '@ember/service';
 import { hash } from 'rsvp';
 import { get, set } from '@ember/object';
 import WithKvActions from 'consul-ui/mixins/kv/with-actions';
-
 export default Route.extend(WithKvActions, {
   templateName: 'dc/kv/edit',
   repo: service('kv'),
-  beforeModel: function() {
+  beforeModel: function(transition) {
+    const url = get(transition, 'intent.url');
+    const search = '/create/';
+    if (url && url.endsWith(search)) {
+      return this.replaceWith('dc.kv.folder', this.paramsFor(this.routeName).key + search);
+    }
```

Comment:
This is "fun" to reason about 😬 - so if it ends in a slash, there's no way someone navigated there from refresh because the binary would cancel it. This is handling the case where they have a key named `create` then?

Reply:
Hehe yeah, it's certainly full of little edge cases.